### PR TITLE
[boost] Modularize

### PR DESCRIPTION
**Pull request by ras0219-msft**
This PR is based off the bincrafter's work to modularize boost.

It uses a powershell script to generate ~130(!) individual packages matching each boost sub-library. The dependencies between these libraries are determined by running `findstr "#include <boost"` over all the sources and headers, with slight massaging.

Some boost header-only components have cyclic dependencies; these have been cut by removing the "weakest" dependency. The logic here is that we treat these "weak" dependencies as optional -- you can install the other packages to light up more features in the header-only library, but it's not required. In the case of circular dependencies between a compiled library and a header only library, the header only library's dependency on the compiled library is removed.

Additionally, the "numeric" library was broken into four sub-libraries: `numeric-conversion`, `interval`, `odeint`, and `ublas`.

There are three "special" boost ports to assist in this:
- The `boost-vcpkg-helpers` port contains this powershell script along with a helper command for the individual portfiles.
- The `boost-build` port builds `b2.exe` and places it into `tools\boost-build\`
- The `boost` port is a metapackage which pulls in every boost port.

Todo:
- [x] Add an `icu` feature to `boost-regex`
- [x] Restrict the `boost` package on `uwp` to only install the available sublibraries
- [x] Ensure that each _compiled_ boost library will build from clean with no other libraries installed.
- [x] Generate appropriate descriptions
- [x] Generate the `boost` metapackage with a suitably advanced version.
- [x] Verify with actual vcpkg users that this is desirable :)
- [x] Reduce existing ports dependencies (required):
  - [x] arrow
  - [x] beast
  - [x] blaze
  - [x] bond
  - [x] cpprestsdk
  - [x] cgal
  - [x] fcl
  - [x] folly
  - [x] hpx
  - [x] libkml
  - [x] libtorrent
  - [x] mongo-cxx-driver
  - [x] openimageio
  - [x] openvdb
  - [x] pcl
  - [x] thrift
  - [x] websocketpp
  - [x] wt
- [x] Ports which will not have dependencies reduced as part of this PR:
  - [x] highfive
  - [x] spirit-po
  - [x] strtk
  - [x] telnetpp
  - [x] think-cell-range

### CODE REVIEW

File: ports/boost-numeric/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,16 @@
+include(vcpkg_common_functions)
+
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO boostorg/numeric
+    REF boost-1.65.1
+    SHA512 Running command:    CertUtil.exe -hashfile "D:\src\vcpkg\ports\boost-modularscripts\downloads\numeric-1.65.1.tar.gz" SHA512  failed
```
Comment (traversaro) [inline ports/boost-numeric/portfile.cmake]: Something went wrong in the automatic generation in this line.

Comment (traversaro) [inline ports/boost-numeric/portfile.cmake]: Something went wrong in the automatic generation in this line.

Comment (ras0219-msft) [reply ports/boost-numeric/portfile.cmake]: Thanks! This was left over from before I split up `boost-numeric` and accidentally slipped in :)

**Comment by pravic** [comment]
Hi there!

As we can see all tasks are resolved. What state of this great rework?

**Comment by ras0219-msft** [comment]
I'm planning to commit it with an upgrade to 1.66 to minimize churn for users; working on that upgrade right now ☺

**Comment by pravic** [comment]
Works smooth. Thank you!

**Pull request merged**